### PR TITLE
refactor(wrapper): refactor the cuda setting

### PR DESCRIPTION
**Pull request by Gaiejj**
## Description

refactor the cuda setting

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [x] I have read the [CONTRIBUTION](https://omnisafe.readthedocs.io/en/latest/developer/contributing.html) guide. (**required**)
- [x] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [x] I have updated the documentation accordingly.
- [ ] I have reformatted the code using `make format`. (**required**)
- [x] I have checked the code using `make lint`. (**required**)
- [x] I have ensured `make test` pass. (**required**)

### CODE REVIEW

File: omnisafe/adapter/offpolicy_adapter.py [hunk 1]
```diff
@@ -61,15 +59,11 @@ def roll_out(  # pylint: disable=too-many-locals
...
-                    torch.rand, size=(self._env.num_envs, *self._env.action_space.shape)
-                )
+            act = torch.as_tensor(self._env.sample_action(), dtype=torch.float32).to(self._device)
         else:
-            act_fn = partial(agent.step, self._current_obs, deterministic=False)
+            act = agent.step(self._current_obs, deterministic=False)
 
         for _ in range(roll_out_step):
-            act = act_fn()
             next_obs, reward, cost, terminated, truncated, info = self.step(act)
```
Comment (rockmagma02) [inline omnisafe/adapter/offpolicy_adapter.py:67]: in all loop of `roll_out_step`, use same `act`? it seem like a bug, we must consider the situation `roll_out_step` != 1

File: omnisafe/envs/safety_gymnasium_env.py [hunk 2]
```diff
@@ -92,7 +92,7 @@ def __init__(self, env_id: str, num_envs: int = 1, **kwargs) -> None:
     def step(
         self, action: torch.Tensor
     ) -> Tuple[torch.Tensor, torch.Tensor, torch.Tensor, torch.Tensor, torch.Tensor, Dict]:
-        obs, reward, cost, terminated, truncated, info = self._env.step(action)
+        obs, reward, cost, terminated, truncated, info = self._env.step(action.cpu().numpy())
```
Comment (rockmagma02) [inline omnisafe/envs/safety_gymnasium_env.py:95]: use `action.detach().cpu().numpy()`

File: omnisafe/envs/core.py [hunk 3]
```diff
@@ -217,16 +218,25 @@ def __getattr__(self, name: str) -> Any:
     def step(
         self, action: torch.Tensor
     ) -> Tuple[torch.Tensor, torch.Tensor, torch.Tensor, torch.Tensor, torch.Tensor, Dict]:
-        return self._env.step(action)
+        obs, reward, cost, terminated, truncated, info = self._env.step(action)
+        return (
```
Comment (rockmagma02) [inline omnisafe/envs/core.py:222]: change device only in `safety_gymnasium_env.py` please, changing device in `Wrapper` could cause multi-calculating in `reset` and `step`

**Review by rockmagma02 — CHANGES_REQUESTED** [review]

File: omnisafe/envs/safety_gymnasium_env.py [hunk 4]
```diff
@@ -75,7 +75,9 @@ class SafetyGymnasiumEnv(CMDP):
     need_auto_reset_wrapper = False
     need_time_limit_wrapper = False
 
-    def __init__(self, env_id: str, num_envs: int = 1, **kwargs) -> None:
+    def __init__(
+        self, env_id: str, num_envs: int = 1, device: torch.device = 'cpu', **kwargs
```
Comment (rockmagma02) [inline omnisafe/envs/safety_gymnasium_env.py:79]: `device: torch.device = torch.device('cpu')`

**Review by rockmagma02 — CHANGES_REQUESTED** [review]

**Review by rockmagma02 — APPROVED** [review]

**Review by zmsn-2077 — APPROVED** [review]
LGTM.
